server-middleware/logHodlers.ts

```diff
@@ -34,73 +72,88 @@ app.post('/logHodlers', async (req: Request, res: Response) => {
 
   // Validates signature sent from client
   const isValid = nacl.sign.detached.verify(encodedMessage, encryptedSignature, publicKey)
-  if(!isValid) {
+  if (!isValid) {
     return res.status(400)
   }
 
-	const discordName = req.body.discordName
-	let tokenList
+  const discordName = req.body.discordName
+  let tokenList
+  let splTokenBalance = 0
 
   // Parses all tokens from that public key
-	try {
-		tokenList = await getParsedNftAccountsByOwner({publicAddress: publicKeyString})
-	} catch (e) {
-		console.log("Error parsing NFTs", e)
-	}
+  try {
+    tokenList = await getParsedNftAccountsByOwner({ publicAddress: publicKeyString })
+  } catch (e) {
+    console.log("Error parsing NFTs", e)
+  }
 
   // Basic ass way to find matched NFTs compared to the mint list ( PRs welcome <3 )
-	let matched = []
-	for (let item of tokenList) {
-		if(mint_list.includes(item.mint)) matched.push(item)
-	}
+  let matched = []
+  for (let item of tokenList) {
+    if (mint_list.includes(item.mint)) matched.push(item)
+  }
+
+  // Optionally check for spl-tokens matching mint IDs if NFTs were not found
+  if (matched.length == 0) {
```

Comment:
I only added this to the first handler to initially add the role. Probably should also update the second one that updates roles later. How does that one get called, didn't see references to it elsewhere in code. Is it meant to be called externally via a `curl` or something?

Reply:
Yeap, curl or just go to it in the browser